apps/scan/backend/package.json

```diff
@@ -19,7 +19,7 @@
     "pre-commit": "lint-staged",
     "start": "node ./build/index.js",
     "test": "is-ci test:ci test:watch",
-    "test:ci": "jest --coverage --reporters=default --reporters=jest-junit --maxWorkers=6",
+    "test:ci": "jest --coverage --reporters=default --reporters=jest-junit --maxWorkers=2 # VxScan backend tests (which involve lots of polling and I/O operations) get flaky when we set --maxWorkers higher",
```

Comment:
My fuzzy theory here is that the VxScan backend tests involve lots of polling and I/O operations (polling because of paper status checks and I/O operations before of continuous export). And setting `--maxWorkers`, i.e. parallelization via worker threads, too high seems to be interfering with those. Also worth noting that worker threads aren't really meant to help with I/O operations. Pulled straight from the [Node docs](https://nodejs.org/api/worker_threads.html):

> Workers (threads) are useful for performing CPU-intensive JavaScript operations. They do not help much with I/O-intensive work. The Node.js built-in asynchronous I/O operations are more efficient than Workers can be.

Our CI build machines seem to have 36 cores (determined by SSH-ing onto one and running `cat /proc/cpuinfo | grep processor | wc -l`), so 6 doesn't feel like it should be a problem. But I SSH-ed onto a build machine and ran some tests. I was able to repro the common failures that we see at 6, 5, and 4 workers. And wasn't able to (in my limited number of test runs) at 3 and below. I landed on 2 to play it safe.

The tradeoff is of course test run time, but even with this change, VxScan backend tests are not the slowest test suite. Not including setup and teardown, the actual test running with 2 workers takes about 90s whereas with 6 workers it took about 45s.

Reply:
Thanks for all this context! Seems reasonable to me